packages/nc-gui/composables/useGlobal/state.ts

```diff
@@ -67,8 +67,9 @@ export function useGlobalState(storageKey = 'nocodb-gui-v2'): State {
   /** saves a reactive state, any change to these values will write/delete to localStorage */
   const storage = useStorage<StoredState>(storageKey, initialState, localStorage, { mergeDefaults: true })
 
+  const route = useRoute()
   /** force turn off of dark mode, regardless of previously stored settings */
-  storage.value.darkMode = false
+  storage.value.darkMode = route.query.dark === '1' || false
 
```

Comment:
Remove this from here and place it into `/plugins/state.ts` 

```ts
// plugins/state.ts

const route = useRoute()
/** if route query is set, use dark mode */
state.darkMode.value = route.query.dark === '1' || false
```

Reply:
changes made, and rebased again